docs/concepts/how-to-grant-api-access.md

```diff
@@ -0,0 +1,183 @@
+  resources:
+  - <your-custom-resources>
+  verbs:
+  - '*'
+```
+
+In each case, replace `<your-api-group>` and `<your-custom-resources>` with the actual API group and resource names provided by the installed operator.
+
+---
+
+## 3. Granting User Access to API Resources
+
+Once the roles are created, you can bind them to specific users or groups to grant them the necessary permissions. There are two main ways to do this:
+
+### Option 1: Binding Default ClusterRoles to Users
+
+- **ClusterRoleBinding**: Use this to grant access across all namespaces.
+- **RoleBinding**: Use this to grant access within a specific namespace.
+
+#### Example: ClusterRoleBinding
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: custom-resource-view-binding
+subjects:
+- kind: User
+  name: <username> # Or use Group for group-based binding
+roleRef:
+  kind: ClusterRole
+  name: custom-resource-view
+  apiGroup: rbac.authorization.k8s.io
+```
+
+This binding grants `<username>` read-only access to the custom resource across all namespaces.
+
+#### Example: RoleBinding
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: custom-resource-edit-binding
+  namespace: <namespace>
+subjects:
+- kind: User
+  name: <username>
+roleRef:
+  kind: Role
+  name: custom-resource-edit
+  apiGroup: rbac.authorization.k8s.io
+```
+
+This RoleBinding restricts permissions to a specific namespace.
+
+### Option 2: Extending Default Kubernetes Roles
+
+To automatically extend existing Kubernetes roles (e.g., the default `view`, `edit`, and `admin` roles), you can add **aggregation labels** to **ClusterRoles**. This allows users who already have `view`, `edit`, or `admin` roles to interact with the custom resource without needing additional RoleBindings.
+
+#### Example: Adding Aggregation Labels to a ClusterRole
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: custom-resource-aggregated-view
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+  - apiGroups:
+      - <your-api-group>
+    resources:
+      - <your-custom-resource>
+    verbs:
+      - get
+      - list
+      - watch
+```
+
+You can create similar ClusterRoles for `edit` and `admin` with appropriate verbs (such as `create`, `update`, `delete` for `edit` and `admin`). By using aggregation labels, the permissions for the custom resources are added to the default roles.
+
+> **Source**: [Kubernetes RBAC Aggregation](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings)
+
+---
+
+## Notes
+
+- OLM does not handle RBAC for users interacting with CRDs, so it's up to cluster administrators to configure these settings.
+- It is not recommended for operator bundles to include RBAC policies granting access to the operator's APIs because cluster administrators should maintain control over the permissions in their clusters.
```

Comment:
Currently, in Kubebuilder, we scaffold only the edit and view RBAC roles, which allows projects to easily provide these roles to their administrators. (For example, see: [Kubebuilder RBAC configuration](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v4/config/rbac)).

_🙏 Your PR highlighted a potential improvement: we could enhance this scaffolding by adding admin roles and including aggregated role annotations. Thanks for bringing this to light—I'll raise an issue to address it._

❓ **On a related note, I have a question**: With the new OLMv1, we’re moving toward a more flexible packaging approach, where we could package various files, such as Helm charts, in the future. Am I right? Then:
- a) Should we recommend users include edit, view, and admin roles in their OLM bundles to allow administrative users more seamless management? 
- b) Is it currently possible or not to include these roles in my OLM bundle?

Reply:
👍 on the issue add upstream.

idk on additional RBAC recommendation because of additional items such as Helm charts, etc. I don't think this doc needs to change for now. We can keep it in mind for once we offer Helm support, etc. Esp. if we see users getting into difficulty in using operators that don't give themselves much RBAC and install these types.

Reply:
```suggestion
- It is recommended that Operator authors include cluster administrative RBAC policies with their solution, as described above. This includes providing view, edit, and admin roles for each API, using the respective Kubernetes annotations. Doing so allows cluster administrators to more easily manage and grant access to the Operator’s APIs for non-administrative cluster users.
```

IHMO, **we should recommend that Operator authors include RBAC rules that allow admins to manage their solutions**. Imagine being a cluster admin with around 50 Operators installed. Would you want to manually configure RBAC rules for each one of the 50 solution—and then recheck them every time there’s an upgrade, since new APIs and permissions might be introduced or modified? Ensuring the correct RBAC rules are in place for management and access on an ongoing basis would create a poor and painful UX for cluster administrators.

Additionally, the guidance above implies the opposite to authors, effectively suggesting, "do not include these rules with your solution because it’s considered a bad practice," which doesn’t seem quite right.

Reply:
The recent commit adds language from the Slack convo. Thanks!

Reply:
resolved with new wording from the slack convo